src/arviz_plots/plots/lm_plot.py

```diff
@@ -0,0 +1,324 @@
+            ci_mean_independent_data = process_group_variables_coords(
+                dt,
+                group="predictions_constant_data",
+                var_names=independent_var,
+                filter_vars=filter_vars,
+                coords=coords,
+            )
+        else:
+            ci_mean_independent_data = independent_data
+    elif isinstance(ci_mean_independent_data, str):
+        if prediction_group == "predictions":
+            pred_independent_var = ci_mean_independent_data
+            ci_mean_independent_data = process_group_variables_coords(
+                dt,
+                group="predictions_constant_data",
+                var_names=ci_mean_independent_data,
+                filter_vars=filter_vars,
+                coords=coords,
+            )
+        else:
+            ci_mean_independent_data = independent_data
+
+    plot_bknd = import_module(f".backend.{backend}", package="arviz_plots")
+    bg_color = plot_bknd.get_background_color()
+    _, contrast_gray_color = get_contrast_colors(bg_color=bg_color, gray_flag=True)
+    if plot_collection is None:
+        pc_kwargs.setdefault("cols", None)
+        pc_kwargs["figure_kwargs"] = pc_kwargs.get("figure_kwargs", {}).copy()
+        pc_kwargs["aes"] = pc_kwargs.get("aes", {}).copy()
+        pc_kwargs = set_wrap_layout(pc_kwargs, plot_bknd, prediction_data)
+        plot_collection = PlotCollection.wrap(
+            prediction_data,
+            backend=backend,
+            **pc_kwargs,
+        )
+
+    if aes_by_visuals is None:
+        aes_by_visuals = {}
+    else:
+        aes_by_visuals = aes_by_visuals.copy()
+    aes_by_visuals.setdefault("line", plot_collection.aes_set)
+
+    # calculations for credible interval
+    if ci_kind == "eti":
+        ci = prediction_data.azstats.eti(prob=ci_prob, **stats.get("credible_interval", {}))
+    elif ci_kind == "hdi":
+        ci = prediction_data.azstats.hdi(prob=ci_prob, **stats.get("credible_interval", {}))
+    ci_upper = ci.y.sel(ci_bound="upper").values
+    ci_lower = ci.y.sel(ci_bound="lower").values
+
+    mean_data = prediction_data.mean(dim=["chain", "draw"])
+    pred_x_vals = ci_mean_independent_data[pred_independent_var].values
+    colors = plot_bknd.get_default_aes("color", 2, {})
+
+    ci_data = np.stack([ci_lower, ci_upper], axis=1)
+
+    if smooth is True:
+        smooth_kwargs = {}
+        smooth_kwargs.setdefault("window_length", 55)
+        smooth_kwargs.setdefault("polyorder", 2)
+        x_data = np.linspace(pred_x_vals.min(), pred_x_vals.max(), 200)
+        x_data[0] = (x_data[0] + x_data[1]) / 2
+        hdi_interp = griddata(pred_x_vals, ci_data, x_data)
+        y_data = savgol_filter(hdi_interp, axis=0, **smooth_kwargs)
+    else:
+        idx = np.argsort(pred_x_vals)
+        x_data = pred_x_vals[idx]
+        y_data = ci_data[idx]
+
+    ci_lower = y_data[:, 0]
+    ci_upper = y_data[:, 1]
+
+    # upper and lower lines of credible interval
```

Comment:
This visual could be named "ci_line" or something similar. Not sure we want this given that we already have "fill", Maybe we want it, but it should be false by default, as we already have "fill". Both lines should be of the same color (maybe constant_gray_color?), and dashed.